app/controllers/application_controller/explorer.rb

```diff
@@ -25,34 +25,10 @@ def try_build_tree(tree_symbol)
     end
   end
 
-  # Historical tree item selected
   def x_history
-    @hist = x_tree_history[params[:item].to_i] # Set instance var so we know hist button was pressed
-    # remove id that is used in replace_right_cell method in vm_common
-    @sb[@sb[:active_accord]] = nil if @sb.fetch_path(@sb[:active_accord]) && @sb[@sb[:active_accord]] != @hist[:id]
-    if @hist[:button] # Button press from show screen
-      self.x_node = @hist[:id]
-      params[:id] = parse_nodetype_and_id(x_node).last
-      params[:x_show] = @hist[:item]
-      params[:pressed] = @hist[:button] # Look like we came in with this action
-      params[:display] = @hist[:display]
-      x_button
-    elsif @hist[:display] # Display link from show screen
-      self.x_node = @hist[:id]
-      params[:id] = parse_nodetype_and_id(x_node).last
-      params[:display] = @hist[:display]
-      show
-    elsif @hist[:action] # Action link from show screen
-      self.x_node = @hist[:id]
-      params[:id] = parse_nodetype_and_id(x_node).last
-      params[:x_show] = @hist[:item]
-      params[:action] = @hist[:action] # Look like we came in with this action
-      session[:view] = @hist[:view] if @hist[:view]
-      send(@hist[:action])
-    else # Normal explorer tree/link click
-      params[:id] = @hist[:id]
-      tree_select
-    end
+    self.x_node = data_for_breadcrumbs[data_for_breadcrumbs.length - 2][:key]
```

Comment:
You should not use `data_for_breadcrumbs` as a source for information. Use the same sources as the method is using, but not the method. 

(Also, this would mean that the method is called 3x on the page.)